### ISSUE TITLE
Could not get meta keys from metastore

### ISSUE THREAD

**Issue by scottyeager** [threefoldtech/0-stor_v2#143]
I noticed while testing the dedup branch at `3accd7` that there are errors like this repeating in the [log file](https://github.com/user-attachments/files/18134014/zstor-dedup.log) after replacing one data and one metadata backend and letting the rebuild operation complete:

```
ERROR Could not get meta keys from metastore: zdb metadata error while: ZDB BACKEND: ZDB at [300:2589:76ee:f44e:a3ad:36d0:8f3c:f297]:9900 504-59257-meta1, error operation READ caused by No: more data
```

It only happens for this one key on this one backend. I checked the connection to the backend and it's fine. There's no obvious issue with the data either:

```console
root@vm:~# redis-cli -h 300:2589:76ee:f44e:a3ad:36d0:8f3c:f297 -p 9900
[300:2589:76ee:f44e:a3ad:36d0:8f3c:f297]:9900> nslist
1) "default"
2) "504-59257-data1"
3) "504-59257-meta1"
[300:2589:76ee:f44e:a3ad:36d0:8f3c:f297]:9900> select 504-59257-meta1 $PASSWORD
OK
[300:2589:76ee:f44e:a3ad:36d0:8f3c:f297]:9900> scan
1) "1\x00\x00\x00\x00\xb2\x02\x00\x00\xc0$\x00\x00=\x16z\x06"
2)  1) 1) "/zstor-meta/meta/26c7cee4bae9b66ab4c0396590460426"
       2) (integer) 330
       3) (integer) 1734136516
    2) 1) "/zstor-meta/meta/404e8b1d07f77ed837dfc793660d80c5"
       2) (integer) 330
       3) (integer) 1734136518
    3) 1) "/zstor-meta/meta/3a49b8691bdba31e7f7923b64fcd8403"
       2) (integer) 330
       3) (integer) 1734136617
    4) 1) "/zstor-meta/meta/323f9ab2a341c7edb4a414eb60397255"
       2) (integer) 330
       3) (integer) 1734136618
    5) 1) "/zstor-meta/meta/e8c5f334c9f39f390bb52006538ef446"
       2) (integer) 570
       3) (integer) 1734136636
    6) 1) "/zstor-meta/meta/10c95b07ca4a0e0966ed2e153167f816"
       2) (integer) 330
       3) (integer) 1734136820
    7) 1) "/zstor-meta/meta/c2d50473cbb7358d452bffb284ac3ea1"
       2) (integer) 330
       3) (integer) 1734136821
    8) 1) "/zstor-meta/meta/adc0536df10097fed133fab8aa35f308"
       2) (integer) 690
       3) (integer) 1734136853
    9) 1) "/zstor-meta/meta/f58da2621af1f25528b9109cc6d9f56f"
       2) (integer) 330
       3) (integer) 1734136854
   10) 1) "/zstor-meta/meta/d4a42488842c95e976974d1e8e98c7dc"
       2) (integer) 690
       3) (integer) 1734136887
   11) 1) "/zstor-meta/meta/e84de87131d307b3679303d4fdc5a407"
       2) (integer) 330
       3) (integer) 1734136887
   12) 1) "/zstor-meta/meta/cb09536e96e41643a4a112bbd8119482"
       2) (integer) 690
       3) (integer) 1734136920
   13) 1) "/zstor-meta/meta/82d142f1ebddd212d7133313d03bcd54"
       2) (integer) 330
       3) (integer) 1734136921
   14) 1) "/zstor-meta/meta/76330048b24beba3320ea59aef554d11"
       2) (integer) 690
       3) (integer) 1734136952
   15) 1) "/zstor-meta/meta/34f1fabbe9936a4626008c1836a6b3c0"
       2) (integer) 330
       3) (integer) 1734136953
   16) 1) "/zstor-meta/meta/d104a2fb60984e5081fb0b76fb077a8b"
       2) (integer) 690
       3) (integer) 1734136983
   17) 1) "/zstor-meta/meta/5a107ef186d4d26c099df33f61d271b4"
       2) (integer) 330
       3) (integer) 1734136983
   18) 1) "/zstor-meta/meta/2ea29feab8dbb939b618bc9e89c6f675"
       2) (integer) 690
       3) (integer) 1734137015
   19) 1) "/zstor-meta/meta/a1c4142a8b326a6e5e121c8bc999613c"
       2) (integer) 330
       3) (integer) 1734137016
   20) 1) "/zstor-meta/meta/35e1718bc3c857b595861d164e5fc49f"
       2) (integer) 690
       3) (integer) 1734137048
   21) 1) "/zstor-meta/meta/46d204eba02568271f9c74394aa2dfec"
       2) (integer) 330
       3) (integer) 1734137048
   22) 1) "/zstor-meta/meta/4287301f433c36be033edd1ca2a645e8"
       2) (integer) 690
       3) (integer) 1734137081
   23) 1) "/zstor-meta/meta/d4354933f80491ae58da09611f903f61"
       2) (integer) 690
       3) (integer) 1734137116
   24) 1) "/zstor-meta/meta/dfb79c97f5739c0aca29549535c76281"
       2) (integer) 330
       3) (integer) 1734137116
   25) 1) "/zstor-meta/meta/6b3de7dad787ac460ef1cd6386e0494a"
       2) (integer) 330
       3) (integer) 1734137117
   26) 1) "/zstor-meta/meta/1465e36182a66d30da559ba45f5f4d1c"
       2) (integer) 690
       3) (integer) 1734137149
   27) 1) "/zstor-meta/meta/45beffd02213cbaa09d4dd00a9dde499"
       2) (integer) 330
       3) (integer) 1734137150
   28) 1) "/zstor-meta/meta/0772b876b7c9030cbc1adc232671f3cd"
       2) (integer) 690
       3) (integer) 1734137183
   29) 1) "/zstor-meta/meta/fb66d28253d16949667da9d998a7c96b"
       2) (integer) 330
       3) (integer) 1734137183
   30) 1) "/zstor-meta/meta/ccbc631a80ecd7707663f29498da3988"
       2) (integer) 690
       3) (integer) 1734137218
   31) 1) "/zstor-meta/meta/0ae2584e9cd8a9d56da0253815ef3a64"
       2) (integer) 330
       3) (integer) 1734137218
   32) 1) "/zstor-meta/meta/6fcb5c7c0b088d8be025986362380c9e"
       2) (integer) 690
       3) (integer) 1734137251
   33) 1) "/zstor-meta/meta/3ea265903052015d542f5688c8744c72"
       2) (integer) 330
       3) (integer) 1734137252
   34) 1) "/zstor-meta/meta/97ca57c2949f38fa167c0327411050d9"
       2) (integer) 690
       3) (integer) 1734137284
   35) 1) "/zstor-meta/meta/a5a1494be9d7fc914cb829813f843ee2"
       2) (integer) 330
       3) (integer) 1734137284
   36) 1) "/zstor-meta/meta/9de91db9a9d5c92195f9a25dc3ea4b20"
       2) (integer) 690
       3) (integer) 1734137317
[300:2589:76ee:f44e:a3ad:36d0:8f3c:f297]:9900> get /zstor-meta/meta/9de91db9a9d5c92195f9a25dc3ea4b20
"\x007f6\x9f\x92\xd8s~\xfbo\xd4\xa4\xd1\x136\xda\x88\x15\xe1K\\8)\x8d^\x06m2\xf9r\xa2Q\xcd7\xf9h\x04>&>\x89G\xe6\xe0\x1b`\xa1\xeb\xf5\a\x0c\xc3R\x9a6\x94\x8cw\xd5C\xc8\xa6\x02e\xa1>o\xd5\xa7\xf1\xaf\x8b\xb6\xfbp6z\xad@+\x98Y\x1b\xc5|%\x8e\x9dW\xfc\xb4t\xa2g4\xe8\xa0\x89Z\xbeI^\xb1.g\x03\xe0\xb00a\xcf@\x15\x9f\x91}p\x9f\x1cu\xb2m\xad84l\x1a,XkB\xb2\xd1:\x14\xd0\xf4\x82\xd3\a`g\xa9\x9b\xefo\xdc\x86\xd7\xc8\x01z#e\x0b\x12\xb2\x93\x81\xb1\xa8\x15\x04\xe8\x19\xd6k\x87T\xcab@\x87\xd9 \x95Y`\x11\xea\xb5\x03\x9a\xee\xa7\xb5\xe0:<l\xa1\x9fI\x9b\xe93\xeeN\xaa\x1e\xcc\xcc\xac\xf3\xe9\xff\x8cne\x92\xe2h\x89a\x99\xda\x16\xa41\x8f\xfc\xa89>EV2\x1e\xc0\xfa&'\xc1v\ns\xc4N\xcd\xff\x1aD\x04+\xe2?qiT\xd30W\x12\xed\xd9*\xa2\x19\xb7\xc4\xae\x1c\x91[\x0c\xb4\a\x01Vvl~/\xcb\xa2\xdc\x1f\x96\x05\xec\x17t\xaeY\xca\xf9\xf4\x80c\xb0\xe2G\xa0\xff\xa4\xff\x93\x1b\x9d\x14\xa1\xd9/PLt\x95\xcf\xcf\xde\xa6M\xef0\xbd\xaa*\x815\x04DH%\x0f\xd0\xfeQ5\xa8\xbb\xd9\xb1\xb8\xd2\xbf\xa1\x92.1Cz\xc9\x19+\n\x13zR\xdb{\\\x8b\xda|i\x11\xa0\xbaPy\x8dYd\xf5 *}&\x8f\xd9$GZ\xde\nwQ\x1cd>yq\xbci\x88\xff:\x91\xfeY2y\xe1\xbb$Z\x05\xa9\x82g\x92<\xe0\x17\x9a\xba\x04\x05\x13\x93\x16C\xbd9\x8d\xcd\x06^\xec^\x9e/\x84=\xec\xe4\xf0g4>\xc2U\x86l\xc2#\xdb\xec\xdf\"\xb0H\x8f\xa6zkL\x90P\xea65;A\x9b5f\x9b\xc4\xbf/?g\r\x1ez\x7fK\xdb\xcab\xea\xfe\xc9\xc5)m%\x9a\xfd\x1e\xef\xcdiD\xb7\xe0\xf2|\xe1E\x93C\x16\xb4\xa6\x15\t\xdelz\xc1=\xda7b\xc7\x85\xf4\r}=`<\x8a\xd0\xce\xb0\x1dj+\xa1\xfe\x162\xf2\x9b\x92\xfc\xe0\x8f4(\x10%\xb0\xd5OD\xd2\b\xbb\x8f\xdd\xb2=ooo\x13\x84\xfd0\x821\xcb\xc6\xba\xe2\xa5\xe9\xa6\x89\x19\xca\x94,\xe5\x0e+!\x88S\xa9)\x86[\xce\x989e\x84\x00\xfb\xbbJ\x18C?\x97R\xec\xc0\xa9U\xb08\xb7&K\t\xc6\xd4\xf3@\xe2[\xbf\x1c\xdd3mPL\xf1\xdf]\xb0\x96\xcd\xdb\xb6[\xdeB\x90\xbb\x90\x05\xc0C\x86G\xd4\xb6\xf8\x83X{yp$D\xac\x88\xda\xc9\xb0Y\xdc\x82 a\x90/\xa1\x93\xc2\xd5\b\x16^\xef\x17\xd5\x9eO\xf7\xeb\xc2\x86(\xae\xaf\xe6\xec\xced\xcdy\xa0;\x85I\x15\xd1\x1ej\xb2"
```

This is a concern if the metadata is in a degraded state for this key and zstor doesn't address it somehow.

**Comment by iwanbk** [comment]
it is actually not really an error because this is what 0-db returned

https://github.com/threefoldtech/0-db?tab=readme-ov-file#scan
```
If the dataset is empty, or you reach the end of the chain, -No more data is returned.
```
I believe the previous method (before improvement) also has this error.

But i agree that the log must be improved because it could be misleading to someone

**Comment by iwanbk** [comment]
PR to fix : #144

**Comment by scottyeager** [comment]
Thanks @iwanbk. I see after checking the logs again that there are success logs for each key, and that this "error" is something extra. On my first read, I thought the error was in place of one success message.

Will be better to have clarity in the logs for sure.
